decomp/semantics/uds.py

```diff
@@ -150,18 +164,30 @@ def from_conll(cls,
         corpus
             (path to) Universal Dependencies corpus in conllu format
         annotations
-            (paths to) annotations in JSON
+            dictionary whose keys are paths to JSON files containing annotations
+            and whose values indicate the type of UDSAnnotation ('raw' or
+            'normalized').
         name
             corpus name to be appended to the beginning of graph ids
         """
-
         predpatt_corpus = PredPattCorpus.from_conll(corpus, name=name)
         predpatt_graphs = {name: UDSGraph(g, name)
                            for name, g in predpatt_corpus.items()}
 
-        annotations = [UDSDataset.from_json(ann) for ann in annotations]
+        # Although neither RawUDSDataset nor NormalizedUDSDataset currently
+        # modifies the from_json method of the abstract base class, we don't
+        # want to assume this, and users should be deliberate in specifying
+        # the annotation type.
+        processed_annotations = []
+        for ann_path, ann_type in annotations.items():
+            if ann_type == 'raw':
+                processed_annotations.append(RawUDSDataset.from_json(ann_path))
+            elif ann_type == 'normalized':
+                processed_annotations.append(NormalizedUDSDataset.from_json(ann_path))
+            else:
+                raise ValueError(f'Unrecognized annotation type {ann_type} for annotation {ann_path}.')
```

Comment:
Use of f-strings requires 3.6. Can you ensure that we require >=3.6? I'm worried that we only require >=3.5, but I could be wrong.

Reply:
The Dockerfile, at least, requires 3.6.